assets/semgrep_rules/client/dangling-pointer-trait.yaml

```diff
@@ -0,0 +1,17 @@
+rules:
+  - id: dangling-pointer-trait
+    metadata:
+      author: Artem Chaikin
+      references:
+        - https://chromium.googlesource.com/chromium/src.git/+/main/docs/dangling_ptr.md
+      source: https://github.com/brave/security-action/blob/main/assets/semgrep_rules/client/dangling-pointer-trait.yaml
+      assignees: |
+        stoletheminerals
+        thypon
+        cdesouza-chromium
+    pattern-regex: |
+      ^.*raw_(ptr|ref)<.*(DanglingUntriaged|DisableDanglingPtrDetection|FlakyDanglingUntriaged|AcrossTasksDanglingUntriaged|AllowPtrArithmetic|AllowUninitialized|LeakedDanglingUntriaged|VectorExperimental|SetExperimental|CtnExperimental).*>.*$
```

Comment:
Does semgrep support generics already or not really?

In case you can use a mix of generic matcher and meta variable match, which would look cleaner

Reply:
.* I think should backtrack and increase the cost of running these, btw

Reply:
Before pushing this I tried metaregex matches in playground, but I couldn't make it work. Even neither of these patterns (with hardcoded trait values) could catch anything in the test file:

```
raw_ptr
raw_ptr...
raw_ptr ...
raw_ptr<...>
raw_ptr<BrowserView, DanglingUntriaged> browser_view_ = nullptr; // the exact string
raw_ptr<$T, DanglingUntriaged> $ARG = $VALUE;
etc
```

Maybe I'm missing something obvious.

Reply:
I'll try quickly. We need to remove the backtracking regex anyway. We should remove any other backtracking if we have too

Reply:
Notice that it could be something like:
```cxx
const raw_ptr<Delegate, AllowPtrArithmetic | FlakyDanglingUntriaged> delegate_;
```
Or

```cxx
const raw_ptr<Delegate, AllowPtrArithmetic 
                                         | FlakyDanglingUntriaged> delegate_;
```

I hope this won't trip the regex

Reply:
removed in https://github.com/brave/security-action/pull/700/commits/54b1aac7814161f9a6573656446f9b7ff1f4c685

Reply:
@cdesouza-chromium addressed in https://github.com/brave/security-action/pull/700/commits/e3492dae259b3b2bee1aae57f3e78cc477a643de